mongorestore/mongorestore_test.go

```diff
@@ -63,6 +68,59 @@ func getRestoreWithArgs(additionalArgs ...string) (*MongoRestore, error) {
 	return restore, nil
 }
 
+func BenchmarkCPUAndMemoryConsumption(b *testing.B) {
+	files, err := ioutil.ReadDir(benchmarkDataRootDir)
+
+	if err != nil {
+		b.Fatalf("Could not find test directory: %v\n", err)
+	}
+
+	for _, file := range files {
+		if file.Mode().IsRegular() {
+			testName := fmt.Sprintf("benchmark for %v", file.Name())
+
+			b.Run(testName, func(b *testing.B) {
+				// Setup arguments and MongoRestore context
+				benchFilePath := filepath.Join(benchmarkDataRootDir, file.Name())
+				collectionName := strings.TrimRight(file.Name(), filepath.Ext(file.Name()))
+
+				args := []string{
+					benchFilePath,
+					DBOption,
+					benchmarkDBName,
+					CollectionOption,
+					collectionName,
+				}
+
+				restore, err := getRestoreWithArgs(args...)
+
+				if err != nil {
+					b.Fatalf("Failed to generate restore context: %v", err)
+				}
+
+				defer func() {
+					client, err := restore.SessionProvider.GetSession()
+					defer restore.Close()
+
+					if err != nil {
+						b.Fatalf("Could not gather context from MongoRestore for %v: %v", testName, err)
+					}
+
+					db := client.Database("benchmark")
```

Comment:
```suggestion
					db := client.Database(benchmarkDBName)
```
[nit] There is a variable for this; use it! (I implied this in my suggestion above)